fuse.py

```diff
@@ -770,11 +794,50 @@ def lock(self, path, fip, cmd, lock):
                                        lock)
 
     def utimens(self, path, buf):
+
+        def is_utime_now(ts):
+            return ts.tv_sec == 0 and ts.tv_nsec == UTIME_NOW
+
+        def is_utime_omit(ts):
+            return ts.tv_sec == 0 and ts.tv_nsec == UTIME_OMIT
+
+        def get_now():
+            if self.flag_nanosecond_int:
+                return time_ns()
+            else:
+                return time()
+
+        def time_of_timespec(ts):
+            if self.flag_nanosecond_int:
+                return int(ts.tv_sec * 1e9 + ts.tv_nsec)
+            else:
+                return ts.tv_sec + ts.tv_nsec * 1e-9
+
         if buf:
-            atime = time_of_timespec(buf.contents.actime)
-            mtime = time_of_timespec(buf.contents.modtime)
+
+            atime_ts = buf.contents.actime
+            mtime_ts = buf.contents.modtime
+
+            atime = time_of_timespec(atime_ts)
+            mtime = time_of_timespec(mtime_ts)
+
+            if self.flag_utime_now_auto:
+                now = get_now()
+                if is_utime_now(atime_ts):
+                    atime = now
+                if is_utime_now(mtime_ts):
+                    mtime = now
+
+            if self.flag_utime_omit_none:
```

Comment:
This looks like it probably should be handled in the `Operations` classes

Reply:
@terencehonles No, really, it should not. Implementing `UTIME_OMIT` and `UTIME_NOW` actually is not optional in POSIX. The only reason I made it optional was because, well, backwards compatibility in the context of current (or past) fusepy. Otherwise, this is critical infrastructure and should not be exposed to or even implemented in the `Operations` class.

Reply:
Thanks for following up, the way you have this has a number of checks that I apparently didn't follow correctly. I mistakenly thought you were only checking for `0` seconds and the flags being set from the features requested (rather than the value of the `tv_nsec`).

So I was under the impression you were adding a non standard feature to the `FUSE` object, not checking to ensure POSIX compliance. I noticed what you were doing and I'm not sure we need to set a feature for that change (the chance that someone sets the `tv_nsec` to the right value accidentally is unlikely).

I'm going to create a PR that you can review, because I do have some questions about this feature.